app/lib/katello/concerns/base_template_scope_extensions.rb

```diff
@@ -129,13 +129,14 @@ def load_errata_applications(filter_errata_type: nil, include_last_reboot: 'yes'
             seen_errata_ids = (seen_errata_ids + parse_errata(task)).uniq
             seen_host_ids << task.input['host']['id'].to_i if include_last_reboot == 'yes'
           end
+          seen_host_ids &= only_host_ids if only_host_ids
 
           # preload errata in one query for this batch
           preloaded_errata = Katello::Erratum.where(:errata_id => seen_errata_ids).pluck(:errata_id, :errata_type)
-          preloaded_hosts = ::Host.where(:id => seen_host_ids & only_host_ids).includes(:reported_data)
+          preloaded_hosts = ::Host.where(:id => seen_host_ids).includes(:reported_data)
 
           batch.each do |task|
-            next unless only_host_ids.include?(task.input['host']['id'].to_i)
+            next unless seen_host_ids && seen_host_ids.include?(task.input['host']['id'].to_i)
```

Comment:
`seen_host_ids` is always array, so we wouldn't need `seen_host_ids &&` part, but I think, we should not change it to `seen_host_ids` here so it should be:
```suggestion
            next unless only_host_ids && only_host_ids.include?(task.input['host']['id'].to_i)
```
1. `seen_host_ids` will be empty array if `include_last_reboot != 'yes'` in such case we would end up with empty report for every `include_last_reboot == 'no'`
2. `seen_host_ids` doesn't need to be complete set even for `include_last_reboot == 'yes'`. Some hosts (hosts without perms, deleted hosts, etc...) may be missing and we still want to include their errata application in the report.

Reply:
Shame on me. Bad conditional.
```suggestion
            next unless only_host_ids.nil? || only_host_ids.include?(task.input['host']['id'].to_i)
```

Reply:
@ezr-ondrej++ PR updated.